### ISSUE TITLE
Maw triggers automatically on un-trashable cards while Imp is installed

### ISSUE THREAD

**Issue by mikeprosser** [mtgred/netrunner#2968]
When accessing a card that can't be stolen or normally trashed (for example, an agenda that has been protected by Lakshmi Smartfabrics or an operation/ice out of a central), Maw triggers immediately on access trashing a card from HQ. The runner can then trigger Imp to trash the card forcing the corp to manually put the card back into HQ, and causing some confusion later if Maw should then resolve further into a turn.

Steps to reproduce
----
1. Have the corp install Lakshmi Smartfabrics, a second asset and a Project Vitruvius, with a second PV in HQ
2. Have the runner install Maw and Imp
3. Make a run on the Vitruvius
4. The corp rezzes Lakshmi, the asset and reveals PV from HQ
5. The runner accesses PV, maw triggers trashing a card from HQ, then the runner can Imp PV

**Comment by danhut** [comment]
I think this needs handling like Mumbad Virtual Tour where if Imp or another "Slow-trash" card is installed we don't trigger the :no-steal event that makes Maw fire.  If the players chooses to NOT use IMP they would need to fire Maw manually.

Maybe... IMP could be changed to :optional so we know when a user has declined to use it?

**Comment by NoahTheDuke** [comment]
Imp is written kinda funny, having come from the agendas. Maybe it's how all the programs are written? I rewrote it, but it doesn't quite work as it needs additional changes elsewhere to properly work:

```clojure
   {:data {:counter {:virus 2}}
    :events {:access {:delayed-completion true
                      :req (req (pos? (get-in card [:counter :virus] 0)))
                      :effect (effect (show-wait-prompt :corp "Runner to use Imp")
                                      (continue-ability
                                        {:optional {:prompt "Trash accessed card at no cost?"
                                                    :once :per-turn
                                                    :yes-ability {:msg "trash at no cost"
                                                                  :effect (effect (add-counter card :virus -1)
                                                                                  (trash-no-cost)
                                                                                  (clear-wait-prompt :corp))}
                                                    :no-ability {:effect (effect (clear-wait-prompt :corp))}}}
                                        card nil))}}}
```

**Comment by NoahTheDuke** [comment]
Closed via #3271.
